common/djangoapps/student/tests/test_views.py

```diff
@@ -778,7 +778,8 @@ def test_content_gating_course_card_changes(self):
         self.override_waffle_switch(True)
 
         course = CourseFactory.create(start=self.THREE_YEARS_AGO)
-        add_course_mode(course, upgrade_deadline_expired=False)
+        add_course_mode(course, mode_slug=CourseMode.AUDIT)
+        add_course_mode(course)
```

Comment:
This and similar changes are because we now check course modes in CourseDurationLimitsConfig.enabled_for_enrollment (like enabled_for_course does and the comments suggest they should).